packages/langium-vscode/src/railroad-webview.ts

```diff
@@ -0,0 +1,120 @@
+/******************************************************************************
+ * Copyright 2023 TypeFox GmbH
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License, which is available in the project root.
+ ******************************************************************************/
+
+import * as vscode from 'vscode';
+import type { LanguageClient } from 'vscode-languageclient/node';
+import { DOCUMENTS_VALIDATED_NOTIFICATION, RAILROAD_DIAGRAM_REQUEST } from './language-server/messages';
+
+export function registerRailroadWebview(client: LanguageClient): void {
+    vscode.commands.registerCommand('langium.showRailroadDiagram', (uri?: vscode.Uri) => {
+        if (uri instanceof vscode.Uri) {
```

Comment:
Can we not use the active document instead?
This case is coming up when one uses the Command list. So there should be a reaction, also when it is not a grammar, like: *Error* can only process *.langium files.